### PR TITLE
Added alias command `automate` for `inspec compliance`

### PR DESCRIPTION
**Pull request by Nik08**
Signed-off-by: Nikita Mathur <nikita.mathur@chef.io>

Added new command alias `automate` for InSpec compliance command 

## Description

Compliance command `automate` introduced for compliance commands, which works similarly to `inspec compliance`.
## Related Issue

Implements #5157 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: docs-chef-io/content/inspec/cli.md [hunk 1]
```diff
@@ -187,6 +187,10 @@ Chef Automate:
 inspec compliance login
 inspec exec compliance://username/linux-baseline
 ```
+An alternate command for login:
+```
+inspec automate login
+```
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/cli.md]:
````suggestion

`inspec automate` is an alias for `inspec compliance` and works the same way:

```
inspec automate login
```
````

File: docs-chef-io/content/inspec/plugin_kitchen_inspec.md [hunk 2]
```diff
@@ -40,6 +40,11 @@ inspec compliance login https://compliance.test --user admin --insecure --token
 
 where `--insecure` is required when using self-signed certificates.
 
+An alternate command for login:
+```bash
+inspec automate login https://compliance.test --user admin --insecure --token ''
+```
+
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/plugin_kitchen_inspec.md]: Someone correct me if I'm wrong, but the reference to Compliance Server starting on line 23 needs to be updated for Automate. I'm not sure if there's a problem with the yaml and bash examples.

````suggestion
`inspec automate` is an alias for `inspec compliance` and works the same way:

```bash
inspec automate login https://compliance.test --user admin --insecure --token ''
```

````

**Review by clintoncwolfe — COMMENTED** [review]
Offhand, the implementation looks OK. However, I am concerned that presenting the `automate` command as an alternative to the `compliance` command is confusing. I wonder if we should do a soft deprecation, presenting it as a replacement while still continuing to support the old command.

**Comment by Nik08** [comment]
@clintoncwolfe So by soft deprecation you do mean that we should show the deprecation warning, right? And yes we are supporting the old command anyway so that will continue if we go for soft deprecation too. 
I personally feel deprecating the old command would be a better experience. Nevertheless, we can take this up with the right team to decide this.

Though, if we present it as an alias rather than an alternative in the docs, does it still seem confusing? 

Also, one of my concern is with the error messages, that's where I need some inputs. Because 'compliance or automate', this part of the error messages does not seem right. You can refer this in target.rb & login.rb in this PR.

File: docs-chef-io/content/inspec/plugin_kitchen_inspec.md (hunk 2 above)
Comment (Nik08) [reply docs-chef-io/content/inspec/plugin_kitchen_inspec.md]: @IanMadd To me it seems all okay. @clintoncwolfe Can you validate this?

File: README.md [hunk 3]
```diff
@@ -177,17 +177,17 @@ You should now be able to run:
...
-  inspec exec PATH(S)                # run all test files at the specified PATH.
-  inspec help [COMMAND]              # Describe available commands or one spe...
-  inspec init TEMPLATE ...           # Scaffolds a new project
-  inspec json PATH                   # read all tests in PATH and generate a ...
-  inspec shell                       # open an interactive debugging shell
-  inspec supermarket SUBCOMMAND ...  # Supermarket commands
-  inspec version                     # prints the version of this tool
+  inspec archive PATH                                       # archive a profile to tar.gz (default) ...
+  inspec check PATH                                         # verify all tests at the specified PATH
+  inspec compliance SUBCOMMAND or automate SUBCOMMAND ...   # Chef Compliance commands
```
Comment (clintoncwolfe) [inline README.md]: List this as:
```
inspec automate SUBCOMMAND # Chef Automate commands
...
inspec compliance SUBCOMMAND # Chef Automate commands (backwards compatible alias)
```

File: dev-docs/compliance.md [hunk 4]
```diff
@@ -65,9 +67,10 @@ There are several other minor commands not listed here - see `lib/cli.rb` for a
 
 ### login
 
-Saves a credentials file locally. Future invocations of `inspec compliance` use the credentials file to authenticate.
+Saves a credentials file locally. Future invocations of `inspec compliance` or `inspec automate` use the credentials file to authenticate.
```
Comment (clintoncwolfe) [inline dev-docs/compliance.md]: Throughout, list `inspec automate` first. Think of it as being the default command, and compliance as being the backwards compatible alias (even though that is not the implementation).

File: dev-docs/compliance.md [hunk 5]
```diff
@@ -65,9 +67,10 @@ There are several other minor commands not listed here - see `lib/cli.rb` for a
...
 
-`be inspec compliance login --user=admin --token='1234567890asdfghjkl' --insecure https://chef-automate.test`
+`be inspec compliance login --user=admin --token='1234567890asdfghjkl' --insecure https://chef-automate.test` or
+`be inspec automate login --user=admin --token='1234567890asdfghjkl' --insecure https://chef-automate.test`
```
Comment (clintoncwolfe) [inline dev-docs/compliance.md]: List automate first.

File: docs-chef-io/content/inspec/cli.md [hunk 6]
```diff
@@ -188,6 +188,12 @@ inspec compliance login
 inspec exec compliance://username/linux-baseline
 ```
 
+`inspec automate` is an alias for `inspec compliance` and works the same way:
```
Comment (clintoncwolfe) [inline docs-chef-io/content/inspec/cli.md]: List automate before compliance, and present compliance as being a backwards-compatible alias.

File: docs-chef-io/content/inspec/plugin_kitchen_inspec.md [hunk 7]
```diff
@@ -40,6 +40,12 @@ inspec compliance login https://compliance.test --user admin --insecure --token
 
 where `--insecure` is required when using self-signed certificates.
 
+`inspec automate` is an alias for `inspec compliance` and works the same way:
```
Comment (clintoncwolfe) [inline docs-chef-io/content/inspec/plugin_kitchen_inspec.md]: List automate before compliance, and present compliance as being a backwards-compatible alias.

**Review by clintoncwolfe — COMMENTED** [review]
I don't think we want to issue a deprecation warning at this point. Mainly I want the wording to be changed so that it appears that automate is the default, and compliance is merely a backwards-compatible alias.

File: dev-docs/compliance.md (hunk 4 above)
Comment (Nik08) [reply dev-docs/compliance.md]: Got it. Will do so. Thanks!

File: lib/plugins/inspec-compliance/lib/inspec-compliance/target.rb [hunk 8]
```diff
@@ -34,13 +34,13 @@ def self.check_compliance_token(uri, config)
         if config["token"].nil? && config["refresh_token"].nil?
           if config["server_type"] == "automate"
             server = "automate"
-            msg = "#{EXEC_NAME} compliance login https://your_automate_server --user USER --ent ENT --dctoken DCTOKEN or --token USERTOKEN"
+            msg = "#{EXEC_NAME} compliance or automate login https://your_automate_server --user USER --ent ENT --dctoken DCTOKEN or --token USERTOKEN"
```
Comment (clintoncwolfe) [inline lib/plugins/inspec-compliance/lib/inspec-compliance/target.rb]: msg = "#{EXEC_NAME} [automate|compliance] login ...

**Review by IanMadd — APPROVED** [review]

File: lib/plugins/inspec-compliance/lib/inspec-compliance/cli.rb [hunk 9]
```diff
@@ -6,8 +6,7 @@ module InspecPlugins
   module Compliance
     class CLI < Inspec.plugin(2, :cli_command)
       include Inspec::Dist
-
-      subcommand_desc "compliance SUBCOMMAND", "#{COMPLIANCE_PRODUCT_NAME} commands"
+      subcommand_desc "automate SUBCOMMAND or compliance SUBCOMMAND", "#{COMPLIANCE_PRODUCT_NAME} commands"
 
       # desc "login https://SERVER --insecure --user='USER' --ent='ENTERPRISE' --token='TOKEN'", 'Log in to a Chef Compliance/Chef Automate SERVER'
       desc "login", "Log in to a #{COMPLIANCE_PRODUCT_NAME}/#{AUTOMATE_PRODUCT_NAME} SERVER"
```
Comment (clintoncwolfe) [inline lib/plugins/inspec-compliance/lib/inspec-compliance/cli.rb]: When you run `inspec automate help`, you see the word `Compliance Server` a lot. I think that is fairly nonsensical now.  Those occurences should be replaced with AUTOMATE_PRODUCT_NAME.

Here, just eliminate the alternative:
`Log in to a #{AUTOMATE_PRODUCT_NAME} SERVER`

Comment (clintoncwolfe) [reply lib/plugins/inspec-compliance/lib/inspec-compliance/cli.rb]: But in each of the command `desc`, let's replace COMPLIANCE_PRODUCT_NAME with AUTOMATE_PRODUCT_NAME.

**Review by clintoncwolfe — COMMENTED** [review]
I've done functional testing on this and it works well. My only concern is the help output from `inspec automate help` continuing to refer to Compliance Server.

**Pull request merged**